### ISSUE TITLE
Architecture build order wrong / xcode

### ISSUE THREAD

**Issue by fhoenig** [shader-slang/slang#2279]
Repro on intel MacOS 12.0.1 / Xcode 13.4.1

`premake5 xcode4 --deps=true --arch=x64`

*open workspace & build*

Error:
`
<path>/slang-dvumxselfvfbmlcasbcsoavvjrfx/Build/Intermediates.noindex/run-generators.build/Script-7017BB282413D45A05C61168.sh: line 5: ../../../bin/macosx-x86/debug/slang-cpp-extractor: No such file or directory
`

`
% ls -l bin/                              
total 0
drwxr-xr-x  3 flo  staff  96 Jun 14 09:12 macosx-aarch64
`

Builds cpp-extractor in arm, but tried to run it on intel.

**Comment by jsmall-nvidia** [comment]
The problem here seems to be something to do with how the 'platform` is handled. 

The reason running the generators fails is that it is looking in the wrong directory for the binaries - it's looking for arm64 binaries. This is even though the project is being built for x86_64. 

This can be worked around by changing the line in `premake5.lua`  

```
platforms { "x86", "x64", "aarch64" }
```

To 

```
platforms {  "x64" }
```

And creating the project with 

```
premake5 xcode4 --deps=true --arch=x64
```

I'm not as yet totally sure how best to fix the problem more appropriately. One way might be to make xcode projects only create a project for the specified arch. Perhaps it's reasonable for xcode project to build for multiple arches, it's not clear then why the arch name doesn't coincide with the target (and directory) name.

**Comment by jsmall-nvidia** [comment]
Fixed on #2300
